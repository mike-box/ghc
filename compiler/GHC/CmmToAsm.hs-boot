module GHC.CmmToAsm
where

import GHC.Prelude

import GHC.IO.Handle.Types
import GHC.CmmToAsm.Config
import GHC.Types.Unique.Supply
import GHC.Utils.Logger
import GHC.Unit.Module.Location
import GHC.Data.Stream
import GHC.Cmm

nativeCodeGen :: forall a . Logger -> NCGConfig -> ModLocation -> Handle -> UniqSupply
              -> Stream IO RawCmmGroup a
              -> IO a
