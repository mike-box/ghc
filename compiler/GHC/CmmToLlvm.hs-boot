module GHC.CmmToLlvm
where

import GHC.Prelude

import GHC.IO.Handle.Types
import {-# SOURCE #-} GHC.CmmToLlvm.Config (LlvmCgConfig)
import GHC.Utils.Logger
import GHC.Data.Stream
import GHC.Cmm

llvmCodeGen :: Logger -> LlvmCgConfig -> Handle
               -> Stream IO RawCmmGroup a
               -> IO a
