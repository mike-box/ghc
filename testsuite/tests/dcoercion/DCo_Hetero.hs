{-# LANGUAGE DataKinds                #-}
{-# LANGUAGE FlexibleInstances        #-}
{-# LANGUAGE GADTs                    #-}
{-# LANGUAGE MultiParamTypeClasses    #-}
{-# LANGUAGE NoImplicitPrelude        #-}
{-# LANGUAGE PolyKinds                #-}
{-# LANGUAGE RankNTypes               #-}
{-# LANGUAGE StandaloneDeriving       #-}
{-# LANGUAGE StandaloneKindSignatures #-}
{-# LANGUAGE Trustworthy              #-}
{-# LANGUAGE TypeFamilies             #-}
{-# LANGUAGE TypeOperators            #-}
{-# LANGUAGE UndecidableInstances     #-}

module DCo_Hetero where

import GHC.Enum
import GHC.Base

type (:~~:) :: k1 -> k2 -> Type
data a :~~: b where
   HRefl :: a :~~: a

instance a ~~ b => Enum (a :~~: b) where
  toEnum = error "toEnum"
