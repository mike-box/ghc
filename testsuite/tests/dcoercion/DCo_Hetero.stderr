
DCo_Hetero.hs:24:10: warning: [-Wmissing-methods (in -Wdefault)]
    • No explicit implementation for
        ‘fromEnum’
    • In the instance declaration for ‘Enum (a :~~: b)’
